### PR TITLE
Add support to use SSL certificate from ACM for TLS termination

### PR DESCRIPTION
**Pull request by dinujoh**
*Issue #, if available:*

*Description of changes:*
* Currently the SSL certificates are read from local file system. This change supports using ACM certificate for TLS. The certificate and encrypted private key are read from the provided ACM ARN and is used to set TLS parameters for the server.
-----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opendistro-for-elasticsearch/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by kowshikn — APPROVED** [review]
LGTM

**Review by chenqi0805 — APPROVED** [review]
LGTM! Thanks for the work.

**Comment by chenqi0805** [comment]
@dinujoh I noticed that the commits are w/o sign-off message. Please doublecheck and do amendment as necessary before merging

File: data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/certificate/acm/ACMCertificateProvider.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        if(exportCertificateResult != null) {
+            final String decryptedPrivateKey = getDecryptedPrivateKey(exportCertificateResult.getPrivateKey(), passphrase);
+            return new Certificate(exportCertificateResult.getCertificate(), decryptedPrivateKey);
+        } else {
+            throw new IllegalStateException(String.format("Exception retrieving certificate results. Time spent retrieving certificate is %d ms and total time out set is %d ms.", timeSlept, totalTimeout));
+        }
+    }
+
+    private String getDecryptedPrivateKey(final String encryptedPrivateKey, final String keyPassword) {
+        Security.addProvider(new BouncyCastleProvider());
```
Comment (wrijeff) [inline data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/certificate/acm/ACMCertificateProvider.java:88]: Does it matter that `addProvider` could be called multiple times (if this method is called multiple times?)

File: data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/certificate/acm/ACMCertificateProvider.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        try {
+            final PrivateKey rsaPrivateKey = encryptedPrivateKeyStringToPrivateKey(encryptedPrivateKey, keyPassword.toCharArray());
+            return privateKeyAsString(rsaPrivateKey);
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
```
Comment (wrijeff) [inline data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/certificate/acm/ACMCertificateProvider.java:93]: Specific exception types to catch? 

I know there's multiple "catch, throw new runtime" throughout the code but we should probably move away from that (eventually, doesn't have to be now).

**Review by wrijeff — APPROVED** [review]
👍 just a few minor comments

Also as @chenqi0805 noted, there's a commit signing requirement [described here](https://github.com/opendistro-for-elasticsearch/data-prepper/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-dco). We could make it more visible to first-time contributors since it's often missed.

**Comment by dinujoh** [comment]
> @dinujoh I noticed that the commits are w/o sign-off message. Please doublecheck and do amendment as necessary before merging

Will amend the commit with sign-off message.

File: data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/certificate/acm/ACMCertificateProvider.java (hunk 2 above)
Comment (dinujoh) [reply data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/certificate/acm/ACMCertificateProvider.java:93]: yeah, makes sense. We can clean this up as part of separate PR.

File: data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/certificate/acm/ACMCertificateProvider.java (hunk 1 above)
Comment (dinujoh) [reply data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/certificate/acm/ACMCertificateProvider.java:88]: Will move this to constructor.

File: data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceSourceConfig.java [hunk 3]
```diff
@@ -64,8 +82,11 @@ public static OTelTraceSourceConfig buildConfig(final PluginSetting pluginSettin
                 pluginSetting.getBooleanOrDefault(PROTO_REFLECTION_SERVICE, false),
                 pluginSetting.getBooleanOrDefault(ENABLE_UNFRAMED_REQUESTS, false),
                 pluginSetting.getBooleanOrDefault(SSL, DEFAULT_SSL),
+                pluginSetting.getBooleanOrDefault(USE_ACM_CERT_FOR_SSL, DEFAULT_USE_ACM_CERT_FOR_SSL),
+                pluginSetting.getLongOrDefault(ACM_CERT_ISSUE_TIME_OUT_MILLIS, DEFAULT_ACM_CERT_ISSUE_TIME_OUT_MILLIS),
```
[the remark this answers is not in the document]
Comment (dinujoh) [reply data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceSourceConfig.java:86]: Will do as part of separate PR.

File: data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceSourceConfig.java [hunk 4]
```diff
@@ -94,6 +115,14 @@ public boolean isSsl() {
         return ssl;
     }
 
+    public boolean isUseAcmCertForSSL() {
```
[the remark this answers is not in the document]
Comment (dinujoh) [reply data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceSourceConfig.java:118]: Will make this change.

**Pull request merged**